server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java

```diff
@@ -84,15 +89,33 @@ public void setProgramName(String programName) {
   }
 
   private List<Variable> getGlobalVariables() {
-    List<Variable> globalVariables = definedVariables.stream().filter(Variable::isGlobal).collect(Collectors.toList());
-    Set<String> globalVariablesNames = globalVariables.stream().map(Variable::getName).collect(Collectors.toSet());
-    List<Variable> parentGlobalVariables = getNearestParentByType(NodeType.PROGRAM)
-        .map(ProgramNode.class::cast)
-        .map(ProgramNode::getGlobalVariables)
-        .orElseGet(ImmutableList::of);
+    List<Variable> globalVariables =
+        definedVariables.stream().filter(Variable::isGlobal).collect(Collectors.toList());
+    Set<String> globalVariablesNames =
+        globalVariables.stream().map(Variable::getName).collect(Collectors.toSet());
+    List<Variable> parentGlobalVariables =
+        getNearestParentByType(NodeType.PROGRAM)
+            .map(ProgramNode.class::cast)
+            .map(ProgramNode::getGlobalVariables)
+            .orElseGet(ImmutableList::of);
     parentGlobalVariables.stream()
         .filter(variable -> !globalVariablesNames.contains(variable.getName()))
         .forEach(globalVariables::add);
     return globalVariables;
   }
+
+  private List<SyntaxError> validateStatements(
+      List<Node> children, Map<Locality, Variable> variables) {
+    List<SyntaxError> result = new ArrayList<>();
+    children.stream()
```

Comment:
Use `getDepthFirstStream` it goes into each element on the tree. You don't need to reimplement it.